run_dim_reduction.py

```diff
@@ -218,56 +226,86 @@ def dim_reduction(input_dirs,
     if method == 'pca':
         fit_func = fit_PCA
         transform_func = process_PCA
     elif method == 'umap':
         fit_func = fit_umap
         transform_func = umap_transform
-        if not fit_model:
+        if not config.dim_reduction.fit_model:
             raise NotImplemented('Inference mode is only supported for PCA at the moment')
     else:
         raise ValueError('Dimensionality reduction method has to be "pca" or "umap"')
-    if conditions is None:
-        conditions = [os.path.basename(input_dir) for input_dir in input_dirs]
-    elif type(conditions) is not list:
-        conditions = [conditions]
-    if fit_model:
+
+    # format conditions as a list
+    if config.dim_reduction.conditions is None:
+        conditions = [os.path.basename(input_dir) for input_dir in config.dim_reduction.input_dirs]
+    elif type(config.dim_reduction.conditions) is not list:
+        conditions = [config.dim_reduction.conditions]
+    else:
+        conditions = config.dim_reduction.conditions
+
+    # run fit for PCA
+    if config.dim_reduction.fit_model:
+        weights_output = os.path.dirname(weights_dir) if os.path.isfile(weights_dir) else weights_dir
         vector_list = []
         labels = []
         label = 0
-        for input_dir in input_dirs:
-            vec = pickle.load(open(os.path.join(input_dir, fname), 'rb'))
-            # vec = zscore(np.squeeze(vec)).astype(np.float32)
-            vector_list.append(vec.reshape(vec.shape[0], -1))
-            labels += [label] * vec.shape[0]
-            label += 1
+
+        # POOL ALL DATA IN INPUT_DIRS
+        for input_dir in config.dim_reduction.input_dirs:
+            # POOL ALL PREFIXES IN FNAME
+            for f in fname:
+                with open(os.path.join(input_dir, f), 'rb') as latent:
+                    vec = pickle.load(latent)
+                # vec = pickle.load(open(os.path.join(input_dir, fname), 'rb'))
+                vector_list.append(vec)
+                labels += [label] * vec.shape[0]
+                label += 1
+
         vectors = np.concatenate(vector_list, axis=0)
-        _ = fit_func(vectors, weights_dir, labels=labels, conditions=conditions)
+        _ = fit_func(vectors, weights_output, labels=labels, conditions=conditions)
         # UMAP model from umap 0.5.0 can't be pickled with protocol=4.
         # Transform from saved models is currently not supported
         if method == 'umap':
             return
+
     # run inference for PCA
-    for input_dir, output_dir in zip(input_dirs, output_dirs):
-        print('Transforming latent vectors for {}'.format(input_dir))
-        transform_func(input_dir=input_dir,
-                    output_dir=output_dir,
-                    weights_dir=weights_dir,
-                    prefix=prefix)
+    if not config.dim_reduction.fit_model:
+        weights_input = os.path.dirname(weights_dir)
+        for input_d, output_d in zip(input_dir, output_dir):
```

Comment:
`input_dirs` and `output_dirs`?

Reply:
actually this is confusing.  Here i say `input_dir` and `output_dir` but it is actually the same as `config.dim_reduction.input_dirs` and `config.dim_reduction.output_dirs`.  I will change the language so it is consistent in this function.

Reply:
`input_dir` here is the last item in config.dim_reduction.input_dirs because it got reassigned in the for loop above. I suggest change back to `input_dirs` or `onfig.dim_reduction.input_dirs`

Reply:
In this implementation of `dim_reduction` this code block would not be accessed simultaneously with the above code block (because `fit_model` has to be `False` here, and `True` for above)